Source/WebCore/html/HTMLMediaElement.cpp

```diff
@@ -6469,6 +6469,11 @@ void HTMLMediaElement::playPlayer()
 #endif
 
     player->play();
+
+#if HAVE(SPEECHRECOGNIZER)
+    if (document().settings().autoGeneratedTranscriptionForVideoEnabled())
+        player->startTranscription();
+#endif
```

Comment:
We don't want to start transcription _every_ time a player starts, instead it should only start when the transcription track created by a player is enabled. The logic to do this will be spread across several classes, so why don't you just remove all of the `startTranscription` code in `HTMLMediaElement` and the `MediaPlayer*` classes for now.

Reply:
I updated the patch with the startTranscription() code removed, but i wonder if it could make sense to call startTranscription when the video is loaded or played for the first time, and within the startTranscription code we could create the synthesized text track and have it disabled by default

Reply:
We talked about this over slack and decided that having startTranscription() be called when the video is loaded would have an impact on power and performance - it would be best to start and stop transcription as the text track is enabled and disabled